### PR TITLE
Add `from __future__ import annotations` in even more files; remove more quotes from annotations

### PR DESCRIPTION
**Pull request by AlexWaygood**
#13391 and #13407 missed a few spots ;)

Cc. @jhance / @cdce8p

### CODE REVIEW

File: mypy/nodes.py [hunk 1]
```diff
@@ -2037,10 +2039,7 @@ class RevealExpr(Expression):
     local_nodes: Optional[List[Var]]
 
     def __init__(
-        self,
-        kind: int,
-        expr: Optional[Expression] = None,
-        local_nodes: "Optional[List[Var]]" = None,
+        self, kind: int, expr: Optional[Expression] = None, local_nodes: Optional[List[Var]] = None
```
Comment (cdce8p) [inline mypy/nodes.py:2042]: Seems like the trailing comma was removed here. I would just add it back

File: mypy/semanal_main.py [hunk 2]
```diff
@@ -108,10 +110,7 @@ def cleanup_builtin_scc(state: "State") -> None:
 
 
 def semantic_analysis_for_targets(
-    state: "State",
-    nodes: List[FineGrainedDeferredNode],
-    graph: "Graph",
-    saved_attrs: SavedAttributes,
+    state: State, nodes: List[FineGrainedDeferredNode], graph: Graph, saved_attrs: SavedAttributes
```
Comment (cdce8p) [inline mypy/semanal_main.py:113]: Here too

**Review by cdce8p — APPROVED** [review]
Looks good AFAICT! Thanks for the follow up @AlexWaygood.
I left two small comments where I think a trailing comma could be useful.

**Comment by AlexWaygood** [comment]
> I left two small comments where I think a trailing comma could be useful.

I tried adding these, but black just strips them away again ;)

**Comment by cdce8p** [comment]
> > I left two small comments where I think a trailing comma could be useful.
> 
> I tried adding these, but black just strips them away again ;)

Found the issue
https://github.com/python/mypy/blob/f8fc26ef76491eae97190e7fb052c695a1a25890/pyproject.toml#L11

The magic-trailing-comma isn't perfect but IMO it works well enough. I would recommend to remove that setting.
https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma

**Comment by AlexWaygood** [comment]
> Found the issue
> 
> https://github.com/python/mypy/blob/f8fc26ef76491eae97190e7fb052c695a1a25890/pyproject.toml#L11

Good catch! I think it was @hauntsaninja that recommended enabling that setting in the initial refactor-for-black PR.

Whatever the wisdom of that setting, however, I'm not going to change that in this PR, since it's slightly tangential to this PR's purpose :-)

**Review by hauntsaninja — APPROVED** [review]

**Comment by hauntsaninja** [comment]
I don't like the magic trailing comma and I would not like that setting changed.

**Comment by cdce8p** [comment]
> I don't like the magic trailing comma and I would not like that setting changed.

It might not be perfect but what I like about it is it at least gives you the option to choose a better formatting if it makes sense. I opened #13418 to show the impact of removing the option.

**Pull request merged**